smoke-test/src/test/java/org/opennms/smoketest/ClassificationRulePageIT.java

```diff
@@ -477,10 +480,20 @@ public String classify(ClassificationRequestDTO classificationRequest) {
             // In some scenarios the response from the opennms endpoint is not yet received, thus the old value is returned
             // resulting in test failures. To avoid this, a bunch of seconds is waited before reading the value.
             // See HZN-1289.
-            sleep(DEFAULT_WAIT_TIME);
 
-            // Fiddle result out of UI
-            return execute(() -> findElementById("classification-response")).getText();
+            String textAfterClassification = "";
+
+            // wait until the classification response value change to a non-empty string that is different from the value
+            // before classification was started
+            for (int i = 0; i < 30; i++) {
+                sleep(DEFAULT_WAIT_TIME);
```

Comment:
You can use  `await().atMost()` pattern here instead of sleep. https://github.com/awaitility/awaitility